satellite/satellitedb/overlaycache.go

```diff
@@ -85,41 +103,91 @@ func (cache *overlaycache) SelectNewStorageNodes(ctx context.Context, count int,
 		args = append(args, v.Major, v.Major, v.Minor, v.Minor, v.Patch)
 	}
 
-	return cache.queryFilteredNodes(ctx, criteria.Excluded, count, safeQuery, args...)
+	var totalNodes []*pb.Node
+	for i := 0; i < 3; i++ {
+		nodes, err := cache.queryFilteredNodes(ctx, criteria.Exclude, count-len(totalNodes), safeQuery, criteria.DistinctIP, args...)
+		if err != nil {
+			return nil, err
+		}
+		for _, n := range nodes {
+			totalNodes = append(totalNodes, n)
+			criteria.Exclude.Nodes = append(criteria.Exclude.Nodes, n.Id)
+			criteria.Exclude.IPs = append(criteria.Exclude.IPs, n.LastIp)
+		}
+		if len(totalNodes) == count {
+			break
+		}
+	}
+	return totalNodes, nil
+}
+
+func (cache *overlaycache) queryFilteredNodes(ctx context.Context, exclude overlay.Exclude, count int, safeQuery string, distinctIP bool, args ...interface{}) (_ []*pb.Node, err error) {
+	switch t := cache.db.DB.Driver().(type) {
```

Comment:
Is there no way around having a selection for each driver here?

Reply:
`DISTINCT ON` is invalid with sqlite